### PR TITLE
Decorated a class

### PR DESCRIPTION
**Pull request by tomasesquivelgc**
# In this branch
 - I create a new nameable class that Person inherits
 - I create a BaseDecorator and Capitalize/Trimmer inherit decorators
 - I successfully implement a Decorator Design

### CODE REVIEW

File: Project/capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative 'base_decorator'
+
+class CapitalizeDecorator < Base_Decorator
```
Comment (rloterh) [inline Project/capitalize_decorator.rb]: - [x] App is not running as intended due to class name typo. Let's change `Base_Decorator` to `BaseDecorator`.

File: Project/trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative 'base_decorator'
+
+class TrimmerDecorator < Base_Decorator
```
Comment (rloterh) [inline Project/trimmer_decorator.rb]: - [x] App is not running as intended due to class name typo. Let's change `Base_Decorator` to `BaseDecorator`.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Tomás, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 🌴  🏃 🌳 🏡 🏘️ 🌲 

## Highlights 👍  
- Good job so far implementing decorator pattern!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Project/capitalize_decorator.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply Project/capitalize_decorator.rb]: Fixed.

File: Project/trimmer_decorator.rb (hunk 2 above)
Comment (Abenezer-Tilahun) [reply Project/trimmer_decorator.rb]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @tomasesquivelgc 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.


## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**